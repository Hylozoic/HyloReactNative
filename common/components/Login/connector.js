import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { login, loginWithFacebook } from './actions'

function mapStateToProps (state) {
  return {
    error: state.loginError,
    defaultEmail: state.defaultLoginEmail
  }
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      login,
      loginWithFacebook
    }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
