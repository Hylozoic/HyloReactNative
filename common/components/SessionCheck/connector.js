import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { pick } from 'lodash/fp'
import { checkSession } from './actions'

const mapStateToProps = pick('loggedIn')

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      checkSession
    }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
