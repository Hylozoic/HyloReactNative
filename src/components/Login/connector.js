import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { login, loginWithFacebook, loginWithGoogle } from './actions'

function mapStateToProps (state) {
  return {
    error: state.session.loginError,
    defaultEmail: state.session.defaultLoginEmail
  }
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      login,
      loginWithFacebook,
      loginWithGoogle
    }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
