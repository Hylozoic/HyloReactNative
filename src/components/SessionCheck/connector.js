import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { checkSession } from './actions'
import getMe from '../../store/selectors/getMe'
import { get } from 'lodash/fp'

export function mapStateToProps (state) {
  const signupInProgress = get('settings.signupInProgress', getMe(state))

  return {
    loggedIn: state.session.loggedIn && !signupInProgress
  }
}

export function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      checkSession
    }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
