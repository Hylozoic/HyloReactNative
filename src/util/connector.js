/**
 * @providesModule util/connector
 */

export const mapWhenFocused = mapper => (stateOrDispatch, props) => props.isFocused
  ? mapper(stateOrDispatch, props)
  : props

export const mergeWhenFocused = merger => (stateProps, dispatchProps, ownProps) =>
  ownProps.isFocused ? merger(stateProps, dispatchProps, ownProps) : ownProps
