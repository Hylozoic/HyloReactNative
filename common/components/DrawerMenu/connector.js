import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { logout } from '../Login/actions'

function mapStateToProps (state) {
  return {} // TODO
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      logout
    }, dispatch)
  }
}

export default connect(mapStateToProps, mapDispatchToProps, null, {withRef: true})
