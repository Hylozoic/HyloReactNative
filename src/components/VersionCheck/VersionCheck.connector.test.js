import { mapStateToProps, mapDispatchToProps } from './VersionCheck.connector.js'

jest.mock('react-native-device-info')

describe('mapStateToProps', () => {
  it('returns the right keys', () => {
    const state = {
      session: {
        loggedIn: true
      },
      pending: {
        CHECK_VERSION: true
      }
    }
    expect(mapStateToProps(state)).toMatchSnapshot()
  })
})

describe('mapDispatchToProps', () => {
  it('maps the action generators', () => {
    const dispatch = jest.fn(val => val)
    const dispatchProps = mapDispatchToProps(dispatch)
    expect(dispatchProps).toMatchSnapshot()
    dispatchProps.checkVersion()
    expect(dispatch).toHaveBeenCalled()
    expect(dispatch.mock.calls).toMatchSnapshot()
  })
})
