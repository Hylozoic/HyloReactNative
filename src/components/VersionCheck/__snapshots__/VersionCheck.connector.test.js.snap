// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapDispatchToProps maps the action generators 1`] = `
Object {
  "checkVersion": [Function],
}
`;

exports[`mapDispatchToProps maps the action generators 2`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "api": Object {
          "method": "get",
          "path": "/noo/mobile/check-should-update?ios-version=2.0",
        },
      },
      "type": "CHECK_VERSION",
    },
  ],
]
`;

exports[`mapStateToProps returns the right keys 1`] = `
Object {
  "pending": true,
  "updateType": Object {
    "type": true,
  },
}
`;
